### ISSUE TITLE
[docx] Update Document Properties in header/footer on open (when no toc)

### ISSUE THREAD

**Issue by agusmba** [jgm/pandoc#5915]
Currently if you use a custom property in your docx reference-doc, in the header/footer part, when you generate a new docx with pandoc, the header/footer will have the property value of the reference-doc, instead of the one you defined in your source document.

If you generate a TOC, Word will ask you about updating everything, so there's no issue if we use TOC, but if we don't, Word doesn't ask you and  you need to manually update them (just going to print-preview in Word updates them for example).

It would be sufficient to add

    <w:updateFields w:val="true" />

to `settings.xml`. However this would trigger for all docx documents, so maybe we could try to get this value from the reference-doc itself. It's a bit of a hassle to have it there since you need to add it at the xml level, I know no way of editing it from within Word itself.

**Comment by jgm** [comment]
Would adding this to all docx documents be a problem?

**Comment by agusmba** [comment]
> Would adding this to all docx documents be a problem?

only in the sense that simple docx documents, without toc, and without fields that need updating on open, will also get the prompt to update them on the first load from Word.

**Comment by jgm** [comment]
Oh, I see.  No, we don't want a prompt.

**Comment by jgm** [comment]
What if we added this ONLY IF the document contains fields?
They'd then be prompted on first load of the document, I assume -- would they always be prompted, or would the prompt go away after first load?

**Comment by mb21** [comment]
> or would the prompt go away after first load?

I think it would only go away if you saved the docx file in word (either overwriting our file or saving a new one), which would be in that (newer) word's file format.

**Comment by agusmba** [comment]
> What if we added this ONLY IF the document contains fields?

That would be great!

> > or would the prompt go away after first load?
> 
> I think it would only go away if you saved the docx file in word (either overwriting our file or saving a new one), which would be in that (newer) word's file format.

Exactly

**Comment by agusmba** [comment]
As a **workaround,** I can manually edit my reference-doc, and add at the xml level in any header/footer with fields the `w:dirty="true"` attribute (at the [right place](http://officeopenxml.com/WPfields.php)).

Since headers and footers are copied completely, the resulting docx will have a dirty flag set and will prompt the user on first Word load.

**Comment by jgm** [comment]
How about this, then?  When we bring the header and footer over from the reference.docx, we check to see if it contains fields.  If it does, we set `w:dirty="true"`.  No change to `settings.xml` would then be needed.

**Comment by jgm** [comment]
@agusmba can you attach or link to a simple reference.docx that has fields in the header or footer?

**Comment by agusmba** [comment]
I uploaded a reference doc and a sample md (remove the txt extension) to show the issue:

[custom-reference.docx](https://github.com/jgm/pandoc/files/3868171/custom-reference.docx)

[pandoc-issue-5915.md.txt](https://github.com/jgm/pandoc/files/3868173/pandoc-issue-5915.md.txt)

```markdown
---
title: Pandoc issue 5915
my-custom-property: 'modified value'
...

Generate a reference doc with:

    pandoc -o custom-reference.docx --print-default-data-file reference.docx

Edit said reference doc to add the `my-custom-property`document property to either
header or footer, with a default value of `reference value`.

Generate a docx from this document with:

    pandoc -o pandoc-issue-5915.docx --reference-doc custom-reference.docx -i pandoc-issue-5915.md

Open the resulting docx, and check whether the custom property has been updated in the 
header/footer: it should show "modified value" (Word should propmt the user about updating
the fields on file open).
```

I don't know which would be easier to implement: marking a field dirty in the header/footer part or adding the `updateFields` at the settings level.

**Comment by jgm** [comment]
Here's the relevant XML:
```
    <w:r>
      <w:rPr>
        <w:lang w:val="es-ES" />
      </w:rPr>
      <w:fldChar w:fldCharType="begin" />
    </w:r>
    <w:r w:rsidRPr="00EB3528">
      <w:instrText xml:space="preserve">
DOCPROPERTY my-custom-property \* MERGEFORMAT
</w:instrText>
    </w:r>
    <w:r>
      <w:rPr>
        <w:lang w:val="es-ES" />
      </w:rPr>
      <w:fldChar w:fldCharType="separate" />
    </w:r>
    <w:r w:rsidRPr="00EB3528">
      <w:t>reference value</w:t>
    </w:r>
    <w:r>
      <w:rPr>
        <w:lang w:val="es-ES" />
      </w:rPr>
      <w:fldChar w:fldCharType="end" />
    </w:r>
```
I tried adding `w:dirty="true"` to the `w:instrText` element, but that didn't do the trick. Where should it go?

**Comment by agusmba** [comment]
According to the [docs](http://officeopenxml.com/WPfields.php) for complex fields it should go in the `fldChar` element that has `fldCharType="begin"` (the docs seem to have a typo since they say *start*).

Something like

```xml
<w:fldChar w:fldCharType="begin" w:dirty="true" />
```

**Comment by jgm** [comment]
So, it should suffice then to scan the header and footer in reference.docx for `w:fldChar` elements with `fldCharType="begin"` and add `w:dirty="true"` whe nwe incorporate these into the document.

**Comment by agusmba** [comment]
Yeah, it should. Even inserting the dirty flag only once should be enough, since Word would update all fields anyway (I think)

**Comment by Dracoriam** [comment]
Just found that it can be done like this. 

```powershell
pandoc -o custom-reference.docx --print-default-data-file reference.docx
```
I added a field:
![image](https://user-images.githubusercontent.com/83291679/129460072-476b7686-5c7a-4b68-82f4-0fa6245f0d84.png)

Insert a field in header
![image](https://user-images.githubusercontent.com/83291679/129459752-a6c56eed-a3a3-48ee-80a0-cb7997b45096.png)
![image](https://user-images.githubusercontent.com/83291679/129459783-ad8dfb98-b6c4-4f04-b94b-5dcfdad4613a.png)
Result
![image](https://user-images.githubusercontent.com/83291679/129459984-8c1a511a-3949-446f-91d8-675b735c9767.png)
Also holds new line, did check anything else.
![image](https://user-images.githubusercontent.com/83291679/129460050-81c443db-40cf-4e2e-a45b-9da29ba4805c.png)

**Comment by Dracoriam** [comment]
Opus. (when no toc) ... Sorry I over reached. The above doesn't work when there is no -toc --table-of-content.
